test/snippet/core/type_list/list_traits_count.cpp

```diff
@@ -0,0 +1,10 @@
+#include <seqan3/core/type_list/traits.hpp>
+
+int main()
+{
+    using t = seqan3::type_list<int, float, bool, int>;
+
+    // Count the number of type int in t.
+    static_assert(std::is_same_v<std::integral_constant<int, 2>,
```

Comment:
why the `integral_constant`?
 Couldn't you just do :
```cpp
seqan3::debug_stream <<  seqan3::list_traits::count<int, t> << "\n"; // prints 2
```
?

Reply:
Hmmm @h-2 wanted me to actually use static_asserts to keep it at compile time instead of debug_streams.

But I realized in some snippets I used debug_streams anyways.

Is it okay if I use debug_streams for some and static_asserts for others? Or should I stick to always using one or the other.

Reply:
Or maybe just 
```cpp
static_assert(seqan3::list_traits::count<int, t> == 2);
```

?

Reply:
:+1: